src/pretix/static/pretixpresale/js/ui/cookieconsent.js

```diff
@@ -0,0 +1,103 @@
+/*global $ */
+
+$(function () {
+    function _send_event(type) {
+        // Event() is not supported by IE11
+        var e = document.createEvent('Event')
+        e.initEvent(type, true, true)
+        document.dispatchEvent(e)
+    }
+
+    if ($("#cookie-consent-storage-key").length) {
+        window.__pretix_cookie_consent = {}
+    } else {
+        window.__pretix_cookie_consent = null
+        return
+    }
```

Comment:
The more I think about it, we could even make `window.pretix.cookie_consent` a function. Also read my comment above, instead of keying by provider, key by usage_class (and maybe additionally provider?). That way, if given only one or two param `key`, returns the cookie-status regarding that `key`. Given three params `window.pretix.cookie_consent('marketing', 'facebook', true)`, it not only sets the value but also emits an update-event. That way the cookie-dialog is basically just another consumer of the `document.addEventListener(pretix:cookie_consent_change)`-event. Setting the whole cookie_consent in one go could also work, as the first param then is not a string, but an object.
To avoid firing multiple upate-events in a row, I would run the update-event after setting the value in a 50ms window.setTimeout.

Of course, if a script wants to set a cookie_consent, it needs to wait for the newly added `document.addEventListener("pretix:cookie_consent:ready")`-event to do so.

Reply:
Or course, we could also just keep it a simple object and do not care about 3rd-parties setting values inside `__pretix_cookie_consent` and not triggering an update.

Reply:
> Also read my comment above, instead of keying by provider, key by usage_class (and maybe additionally provider?).

Our legal counsel says people need to be able to turn off individual providers, so I do not think keying of provider gives us anything, it just makes things harder.